### ISSUE TITLE
[caddy 1.0.0][go1.12.4] Can't build with caddy's latest source code

### ISSUE THREAD

**Issue by IceCodeNew** [pyed/ipfilter#41]
Firstly, I tried to get the officially built caddy with http.ipfilter plugged in: https://caddyserver.com/download/linux/amd64?plugins=http.ipfilter&license=personal&telemetry=off
Which returned such result: ``` provisioning build environment: go get: exit status 1 ```

Then I Tried to build caddy from source with ipfiler in a brand new VPS environment:
// I was following the official guide here: https://github.com/mholt/caddy/wiki/Plugging-in-Plugins-Yourself

\# go env
```
GOARCH="amd64"
GOBIN=""
GOCACHE="/root/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/root/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/root/go/caddy/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build112826577=/tmp/go-build -gno-record-gcc-switches"
```

\# export GO111MODULE=on

\# vim run.go
```
package main
import (
    "github.com/mholt/caddy/caddy/caddymain"

    // plug in plugins here, for example:
    // _ "import/path/here"
    _ "github.com/pyed/ipfilter"
)
func main() {
    // optional: disable telemetry
    caddymain.EnableTelemetry = false
    caddymain.Run()
}
```

\# go mod init caddy

\# go build
```
# bytes
/usr/local/go/src/bytes/bytes_decl.go:10:6: IndexByte redeclared in this block
        previous declaration at /usr/local/go/src/bytes/bytes.go:101:34
/usr/local/go/src/bytes/bytes_decl.go:17:6: Equal redeclared in this block
        previous declaration at /usr/local/go/src/bytes/bytes.go:18:25
/usr/local/go/src/bytes/bytes_decl.go:24:6: Compare redeclared in this block
        previous declaration at /usr/local/go/src/bytes/bytes.go:37:27
# strings
/usr/local/go/src/strings/strings_decl.go:8:6: IndexByte redeclared in this block
        previous declaration at /usr/local/go/src/strings/strings.go:150:34
# crypto/cipher
/usr/local/go/src/crypto/cipher/xor_amd64.go:9:6: xorBytes redeclared in this block
        previous declaration at /usr/local/go/src/crypto/cipher/xor.go:58:33
/usr/local/go/src/crypto/cipher/xor_amd64.go:22:6: xorWords redeclared in this block
        previous declaration at /usr/local/go/src/crypto/cipher/xor.go:83:27
# crypto/rc4
/usr/local/go/src/crypto/rc4/rc4_asm.go:15:18: (*Cipher).XORKeyStream redeclared in this block
        previous declaration at /usr/local/go/src/crypto/rc4/rc4.go:61:6
```

When the above commands finished I found no binary file had been generated, any idea?

**Comment by mholt** [comment]
Additionally, I believe the filenames in `testdata` folder that have colons in them (`::1`) is causing builds to fail as well. (I'm seeing errors in our build logs about that.)

**Comment by krader1961** [comment]
> Additionally, I believe the filenames in testdata folder that have colons in them (::1) is causing builds to fail as well.

That was changed four days ago with commit ba3fef8c6. See issue #39.

**Comment by mholt** [comment]
Oh, excellent! Thank you.

**Comment by pyed** [comment]
I just published and deployed a new version `v1.1.3` that uses the new go modules, try building again using the same instructions you linked, it worked for me.

**Comment by IceCodeNew** [comment]
Yes, the latest version works for me too. Thanks a lot!
